### PR TITLE
feat(robot-server): http protocol upload supports custom labware.

### LINKED ISSUE #7148: feat(robot-server,api): Custom labware support in HTTP protocol sessions
# Overview 

HTTP protocol upload and protocol sessions provide limited support for custom labware.  Just as python protocols uploaded by RPC, HTTP API uploaded protocols need to support loading custom labware in `ProtocolContext.load_labware`.

The runapp uploads all custom labware when creating an RPC protocol session. These labware definitions are supplied to `ProtocolContext` as `extra_labware` member: a mapping of load name to custom labware definition.

# Implementation

- A custom labware definition is a JSON file that the user uploads as a protocol "support file"

# Acceptance Criteria

- Protocols can use `ProtocolContext.load_labware` to load a labware from a definition stored in a support JSON file.

### PR DESCRIPTION
**Pull request by amitlissack**
# Overview

Protocol upload supports custom labware for python protocols. 

A custom labware is a json file that is included as a support file. 

At analysis time we use `entrypoint_util.labware_from_paths` to find valid labware files in the protocol temp directory. The discovered labware are supplied to the `ProtocolContextImplementation` as `extra_labware`. 

At run time we again use `entrypoint_util.labware_from_paths` and supply the results to `Session.build_and_prep` as a list. 

closes #7148 

# Changelog

- search for custom labware using `entrypoint_util.labware_from_paths` 
- remove redundant unit tests
- split up protocol tavern tests into multiple files.
- add tavern tests for custom labware.

# Review requests

I don't believe this is the long term solution for custom labware, but it works and requires the least amount of code changes.

In the future, as we work on protocol runner, I see us having a `LabwareResolver` type class that can lazily load definitions. 
  
# Risk assessment

Very little.

### CODE REVIEW

File: robot-server/tests/integration/protocols/test_support_files.tavern.yaml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                pipetteName: p300_single
+                channels: 1
+            labware:
+              - label: corning_96_wellplate_360ul_flat
+                uri: opentrons/corning_96_wellplate_360ul_flat/1
+                location: 1
+              - label: opentrons_96_tiprack_300ul
+                uri: opentrons/opentrons_96_tiprack_300ul/1
+                location: 2
+              - !anydict
```
Comment (SyntaxColoring) [inline robot-server/tests/integration/protocols/test_support_files.tavern.yaml]: What's this `!anydict` for?

**Review by SyntaxColoring — COMMENTED** [review]
Neat! Most of this is over my head—maybe we should go over it in pairing.

File: robot-server/tests/integration/protocols/test_support_files.tavern.yaml (hunk 1 above)
Comment (amitlissack) [reply robot-server/tests/integration/protocols/test_support_files.tavern.yaml]: I only care about labware in slots 1 and 2. There will be labware in the trash slot. I'll add comment.

**Review by sfoster1 — APPROVED** [review]
Nice, looks good to me!

**Review by b-cooper — APPROVED** [review]
🥼 looks like a totally acceptable solution. Lazy loading will be a nice evolution of this later on.

**Review by mcous — APPROVED** [review]
Oops, too late. Original review below

---

I think this is a smart approach 👍

**Pull request merged**